### PR TITLE
[CIS-550] MessageComposer basic animation

### PR DESCRIPTION
**Pull request by mantergo**
**In this PR:**

- Update `textColor` with `ColorPalette`
- Fix sendting attachments without text
- Remove left and right extra insets for `MessageComposerView`
- Add basic animation to `MessageComposer`

_Note: haven't succeeded in animating intrinsicContentSize change on textView height expand._

https://user-images.githubusercontent.com/10098359/103308065-d9912680-4a11-11eb-9fe6-c75591f47073.mp4

### CODE REVIEW

**Review by VojtaStavik — CHANGES_REQUESTED** [review]
Great start. I think it'd be nice if we had just once place where to perform animations so we can control is centrally.

Maybe something like: 
```swift
func WithAnimation(_ actions: () -> Void) {...}
```
Inside this function, we should also check for `UIAccessibilityIsReduceMotionEnabled` value and don't perform animations if it's set.

Finally, let's try to use a spring animation as default. The UI will feel much more snappier.

**Comment by mantergo** [comment]
> Great start. I think it'd be nice if we had just once place where to perform animations so we can control is centrally.
> 
> Maybe something like:
> 
> ```swift
> func WithAnimation(_ actions: () -> Void) {...}
> ```
> 
> We'd use it as replacement for `UIView.animate { }`. Inside this function, we should also check for `UIAccessibilityIsReduceMotionEnabled` value and don't perform animations if it's set.
> 
> Finally, let's try to use a spring animation as default. The UI will feel much more snappier.

Yeah, good points! Thanks!

File: Sources_v3/StreamChatUI/Utils/Animation.swift [hunk 1]
```diff
@@ -0,0 +1,22 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import UIKit
+
+func WithAnimation(_ actions: @escaping () -> Void, completion: ((Bool) -> Void)? = nil) {
```
Comment (VojtaStavik) [inline Sources_v3/StreamChatUI/Utils/Animation.swift:7]: nit: Are you happy with the name? It seems I don't actually like it 🤦‍♂️ Maybe `Animate { }` would read better on the caller side?

File: Sources_v3/StreamChatUI/Utils/Animation.swift [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            usingSpringWithDamping: 0.5,
+            initialSpringVelocity: 4,
+            options: .curveEaseInOut,
+            animations: actions,
+            completion: completion
+        )
+    } else {
+        actions()
+        completion?(true)
+    }
```
Comment (VojtaStavik) [inline Sources_v3/StreamChatUI/Utils/Animation.swift:21]: nit: `guard` would work much better here

**Review by VojtaStavik — APPROVED** [review]

**Pull request merged**